firmware/net/components/wifi/wifi.cc

```diff
@@ -26,125 +31,168 @@ Wifi::~Wifi()
 
 void Wifi::Begin()
 {
-    connect_semaphore = xSemaphoreCreateBinary();
-
-    // Start with a value of 1 for readability
-    xSemaphoreGive(connect_semaphore);
-
-    ESP_ERROR_CHECK(Initialize());
-
-    xTaskCreate(ConnectTask,
+    xTaskCreate(WifiTask,
         "wifi_connect_task",
         4096,
         (void*)this,
         12,
         &connect_task);
 }
 
+const std::vector<std::string>& Wifi::GetNetworksInRange() const
+{
+    return scanned_aps;
+}
+
+void Wifi::Connect(const std::string& ssid, const std::string& pwd)
+{
+    if (ssid.length() == 0 || pwd.length() == 0)
+    {
+        return;
+    }
+
+    ap_cred_t creds = { ssid, pwd, 0 };
+    credentials.push_back(std::move(creds));
+}
+
 void Wifi::Connect(const char* ssid, const char* password)
 {
-    NET_LOG_ERROR("ssid len", std::min(strlen(ssid), (size_t)sizeof(wifi_cfg.sta.ssid)));
-    NET_LOG_ERROR("ssid pwd len", std::min(strlen(password), (size_t)sizeof(wifi_cfg.sta.password)));
-    Connect(ssid,
-        std::min(strlen(ssid), (size_t)sizeof(wifi_cfg.sta.ssid)),
-        password,
-        std::min(strlen(password), (size_t)sizeof(wifi_cfg.sta.password))
-    );
+    Connect(std::string(ssid), std::string(password));
```

Comment:
```suggestion
    Connect(ssid, password);
```
C++ will implicitly construct strings for you, but there's no harm in being explicit either.

Reply:
Didn't even realise this was inside another Connect method, the std::string one should be the only one